packages/app/src/templates/services/template.service.ts

```diff
@@ -42,4 +49,48 @@ export class TemplateService {
       }
     }
   }
+
+  public async deleteTemplates(
+    templateIds: string[]
+  ): Promise<RequestResponse<string[]>> {
```

Comment:
What I found problematic here:
1. error response is not typed
2. we need to examine the function to understand what `string[]` means.

Maybe we could go with such solution:
1. we could make `RequestResponseError` to accept generic type for `data`
```
export interface RequestResponseError<ErrorDataType = any> {
  code?: number
  message: string
  data?: ErrorDataType
}

export interface RequestResponse<DataType = undefined, ErrorDataType = any> {
  status: RequestResponseStatus
  data?: DataType
  error?: RequestResponseError<ErrorDataType>
}
``` 
And later you could go with:
```
type DeletedTemplatesIds = string[] 
type ErrorTemplatesIds = string[]
// above types make code reader less confused - he/she immediately knows what is returned when error or success appears

type DeleteTemplateRequestResponse = RequestResponse<DeletedTemplatesIds, ErrorTemplatesIds>
```
and reuse in whenever you need it (e.g. `template.service.ts`, `delete-templates.request.ts` etc.). By that we are a little bit safer - cause we are protected by types.

What do you think about it?

Reply:
Looks like a nice improvement, we can use it also in other request responses :) 
Changed in [5ca7841](https://github.com/mudita/mudita-center/pull/752/commits/5ca7841b5759555b1bd575843e9a4c7a8ea57487)